keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Variable.java

```diff
@@ -1,10 +1,16 @@
 package io.improbable.keanu.algorithms.variational.optimizer;
 
+import io.improbable.keanu.network.VariableState;
+
 public interface Variable<T> {
 
     VariableReference getReference();
 
     T getValue();
 
     long[] getShape();
+
+    VariableState getState();
+
+    void setState(VariableState variableState);
```

Comment:
I was hoping to keep Variable very functionally limited. Putting set here opens some complicated questions around setting and cascading or just setting. If I do a set on this variable does it affect other variables? If not, then this might get confusing and if so then this implies some sort of value propagation algorithm operating on the new state, which is basically the Vertex object we already have.